pipeline/02-assess.R

```diff
@@ -43,20 +43,49 @@ lgbm_final_full_recipe <- readRDS(paths$output$workflow_recipe$local)
 # FMV per unit
 assessment_data_pred <- read_parquet(paths$input$assessment$local) %>%
   as_tibble() %>%
-  mutate(
-    pred_card_initial_fmv = predict(
-      lgbm_final_full_fit,
-      new_data = bake(
-        lgbm_final_full_recipe,
-        new_data = .,
-        all_predictors()
-      )
-    )$.pred
-  )
+  # Bake the data first and extract meta strata columns
+  {
+    baked_data <- bake(lgbm_final_full_recipe, new_data = ., all_predictors())
+    mutate(
+      .,
+      pred_card_initial_fmv = as.numeric(predict(
+        lgbm_final_full_fit,
+        new_data = baked_data
+      )$.pred),
+      # Some strata are imputed during the baking process
+      # so we extract all values.
+      temp_strata_1 = baked_data$meta_strata_1,
+      temp_strata_2 = baked_data$meta_strata_2
+    )
+  }
 
+# For the lightgbm model, values are recoded to a 0 based scale.
+# This means that these values are a 1:1 match with values of a
+# different scale. Because of this, we map values to our original
+# calculations for continuity.
+mapping_1 <- assessment_data_pred %>%
+  filter(!is.na(meta_strata_1)) %>%
+  distinct(temp_strata_1, meta_strata_1)
 
+mapping_2 <- assessment_data_pred %>%
+  filter(!is.na(meta_strata_2)) %>%
+  distinct(temp_strata_2, meta_strata_2)
 
+strata_mapping_1 <- setNames(mapping_1$meta_strata_1, mapping_1$temp_strata_1)
+strata_mapping_2 <- setNames(mapping_2$meta_strata_2, mapping_2$temp_strata_2)
```

Comment:
**nitpick:** You can clean this up a tiny bit using `set_names` from `rlang` or `purrr`:

```r
strata_mapping_2 <- assessment_data_pred %>%
  filter(!is.na(meta_strata_2)) %>%
  distinct(meta_strata_2, temp_strata_2) %>%
  purrr::set_names(.$meta_strata_2, .$temp_strata_2)
```

Reply:
This gives me the error message

![image](https://github.com/user-attachments/assets/88dea559-6628-416a-9c5d-f50fc30bcf46)

I tried something different:
```
strata_mapping_1 <- assessment_data_pred %>%
  filter(!is.na(meta_strata_1)) %>%
  mutate(meta_strata_1 = as.character(meta_strata_1)) %>%
  distinct(temp_strata_1, meta_strata_1) %>%
  mutate(across(c(temp_strata_1, meta_strata_1))) %>%
  pull(temp_strata_1) %>%
  set_names(
    assessment_data_pred %>%
      filter(!is.na(meta_strata_1)) %>%
      distinct(temp_strata_1, meta_strata_1) %>%
      pull(meta_strata_1)
  )
```
This recodes 0 in unreliable ways. You can manually recode from 0 to something else, but that seems way less stable. See the reproduced table for an example.
![image](https://github.com/user-attachments/assets/46bcd50c-74be-47db-b539-edbfac7c23c8)